code/src/java/pcgen/gui2/facade/CharacterFacadeImpl.java

```diff
@@ -373,7 +375,18 @@ private void initForCharacter()
 		kitList = new DefaultListFacade<KitFacade>();
 		refreshKitList();
 
-		statScoreMap = new HashMap<StatFacade, DefaultReferenceFacade<Integer>>();
+		statScoreMap = new HashMap<StatFacade, WriteableReferenceFacade<Integer>>();
+		for (StatFacade stat : dataSet.getStats())
+		{
+			if (stat instanceof PCStat)
+			{
+				statScoreMap.put(stat, StatAdapter.generate(theCharacter.getCharID(), (PCStat) stat));
+			}
+			else
+			{
+				statScoreMap.put(stat, new DefaultReferenceFacade<Integer>());
```

Comment:
Tactically, I think the checkin is okay, but strategically, this is the one part that stands out to me.  My personal preference would be some form of message here since we should assert that StatFacade == PCStat.  If we don't make that assumption, then we really should be putting StatAdapter.generate into DataSetFacade so that it's hidden from CharacterFacadeImpl? ... or what's the appropriate place so that the assumption of StatFacade == PCStat is always in one place?